src/library/scala/collection/Iterable.scala

```diff
@@ -332,8 +332,8 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
         val thisIt = this.iterator
         val thatIt = that.iterator
         while (thisIt.hasNext && thatIt.hasNext) {
-          thisIt.next()
-          thatIt.next()
+          thisIt.next(): Unit
+          thatIt.next(): Unit
```

Comment:
The other random syntax idea: `(thisIt.next())`, i.e., don't warn about expressions that seem to be embedded in the literal unit `()` parens.